packages/geometry/src/rectangle.ts

```diff
@@ -206,6 +206,26 @@ export function pad(rect: Rectangle, padding: number): Rectangle {
   );
 }
 
+/**
+ * Returns `true` if the given rectangle contains all the given `points`.
+ *
+ * @param rect The rectangle to check against.
+ * @param points The points to check.
+ */
+export function containsPoints(
+  rect: Rectangle,
+  ...points: Point.Point[]
+): boolean {
+  return points.every((point) => {
+    return (
+      rect.x < point.x &&
+      rect.x + rect.width > point.x &&
```

Comment:
if `rect.x == 0` and `rect.width == 1`, neither `point.x==0` nor `point.x==1` satisfy this condition.  I don't think that's what you want or expect.

What we've done before is define it has a half-interval, which leads to logic like `rect.x <= x && rect.x+rect.width < x`.  So `x==0` would pass but `x==1` would not.

Reply:
Ended up modifying this to allow for both the `point.x === 0` and `point.x === 1` case, particularly for the case I'm using this for - which is determining whether a point on screen is within the 2D bounds of the transform widget so we can start performing hit-tests for the hovered state. Open to renaming this, or adding some additional parameter to specify the behavior though

Reply:
The logic you have here only accepts points "in" the rect but not "on" the rect.  That's fine if that's what you need but it is a subtle distinction that should probably be documented somewhere.